test/cache/DiskBasedPackageCache.test.ts

```diff
@@ -0,0 +1,164 @@
+    });
+  });
+
+  describe('#getPackagePath', () => {
+    it('should return the package path for a package in the cache', () => {
+      expect(cache.getPackagePath('fhir.small', '0.1.0')).toBe(
+        path.resolve(cacheFolder, 'fhir.small#0.1.0')
+      );
+    });
+
+    it('should return undefined for a package not in the cache', () => {
+      expect(cache.getPackagePath('big', '0.1.0')).toBeUndefined();
+    });
+
+    it('should return undefined for a package with different version in the cache', () => {
+      expect(cache.getPackagePath('fhir.small', '0.2.0')).toBeUndefined();
+    });
+  });
+
+  describe('#getPackageJSONPath', () => {
+    it('should return the path to package.json for a package in the cache', () => {
+      expect(cache.getPackageJSONPath('fhir.small', '0.1.0')).toBe(
+        path.resolve(cacheFolder, 'fhir.small#0.1.0', 'package', 'package.json')
+      );
+    });
+
+    it('should return undefined for a package not in the cache', () => {
+      expect(cache.getPackageJSONPath('big', '0.1.0')).toBeUndefined();
+    });
+
+    it('should return undefined for a package with different version in the cache', () => {
+      expect(cache.getPackageJSONPath('fhir.small', '0.2.0')).toBeUndefined();
+    });
+  });
+
+  describe('#getPotentialResourcePaths', () => {
+    it('should return potential paths for a package in the cache', () => {
+      const potentials = cache.getPotentialResourcePaths('fhir.small', '0.1.0');
+      expect(potentials).toHaveLength(3);
+      const rootPath = path.resolve(cacheFolder, 'fhir.small#0.1.0', 'package');
+      expect(potentials).toContain(path.resolve(rootPath, 'ImplementationGuide-fhir.small.json'));
+      expect(potentials).toContain(path.resolve(rootPath, 'package.json'));
+      expect(potentials).toContain(path.resolve(rootPath, 'StructureDefinition-MyPatient.json'));
+      expect(loggerSpy.getAllLogs()).toHaveLength(0);
+    });
+
+    it('should return empty array for a package not in the cache', () => {
+      const potentials = cache.getPotentialResourcePaths('big', '0.1.0');
+      expect(potentials).toHaveLength(0);
+      expect(loggerSpy.getAllLogs()).toHaveLength(0);
+    });
+
+    it('should return empty array for a package with different version in the cache', () => {
+      const potentials = cache.getPotentialResourcePaths('fhir.small', '0.2.0');
+      expect(potentials).toHaveLength(0);
+      expect(loggerSpy.getAllLogs()).toHaveLength(0);
+    });
+  });
+
+  describe('#getResourceAtPath', () => {
+    it('should return a resource with a given resource path', () => {
+      const rootPath = path.resolve(cacheFolder, 'fhir.small#0.1.0', 'package');
+      const totalPath = path.resolve(rootPath, 'StructureDefinition-MyPatient.json');
+      const resource = cache.getResourceAtPath(totalPath);
+      expect(resource).toBeDefined();
+      expect(resource.id).toBe('MyPatient');
+      expect(loggerSpy.getAllLogs('error')).toHaveLength(0);
+    });
+  });
```

Comment:
In theory, this would only happen if someone deleted the file after it was loaded but before it was queried. That said, I implemented specific behavior for this case, so we should test it.